java_package/brainflow/src/main/java/brainflow/NativeLoader.java

```diff
@@ -0,0 +1,44 @@
+package brainflow;
+
+import com.sun.jna.Platform;
+
+import java.io.File;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+class NativeLoader
+{
+    static Path unpack_from_jar (String lib_name)
+    {
+        try
+        {
+            File file = new File (lib_name);
+            if (file.exists ())
+                file.delete ();
+            InputStream link = (NativeLoader.class.getResourceAsStream (lib_name));
+            Files.copy (link, file.getAbsoluteFile ().toPath ());
+            return file.getAbsoluteFile ().toPath ();
+        } catch (Exception io)
+        {
+            System.err.println ("file: " + lib_name + " is not found in jar file");
+            return null;
+        }
+    }
+
+    static void copy_to_temp_dir (String lib_name)
+    {
+        // https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/MacOSXDirectories/MacOSXDirectories.html
+        File jnatmp = new File (System.getProperty ("user.home"), "Library/Caches/JNA/temp/" + lib_name);
```

Comment:
I dont like hardcode for jna temp dir, there is a method for it https://github.com/java-native-access/jna/blob/bf60e51eace6dffa18548019e2ba398ff84904ef/src/com/sun/jna/Native.java#L1321 and its OS agnostic